docs/proposals/container-runtime-interface-v1.md

```diff
@@ -0,0 +1,182 @@
+# Redefine Container Runtime Interface
+
+## Motivation
+
+Kubelet employs a declarative pod-level interface, which acts as the sole
+integration point for container runtimes (e.g., `docker` and `rkt`). The
+high-level, declarative interface has caused higher integration and maintenance
+cost, and also slowed down feature velocity for the following reasons.
+  1. **Not every container runtime supports the concept of pods natively**.
+     When integrating with Kubernetes, a significant amount of work needs to
+     go into implementing a shim of significant size to support all pod
+     features. This also adds maintenance overhead (e.g., `docker`).
+  2. **High-level interface discourage code sharing and reuse among runtimes**.
+     E.g, each runtime implements an all-encompassing `SyncPod()` function that
```

Comment:
nit: It is not clear that `SyncPod()` is a declarative API from this line. Can you add a few phrases to make that explicit?